subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/report/AbstractTablePageGenerator.java

```diff
@@ -230,12 +230,12 @@ private void renderDetailsTable(PerformanceReportScenario scenario) {
                     tr();
                         th().text("Date").end();
                         th().text("Commit").end();
-                        renderVersionHeader(scenario.getRecentExecutions().isEmpty() ? "" : scenario.getRecentExecutions().get(0).getBaseVersion().getName());
-                        renderVersionHeader(scenario.getRecentExecutions().isEmpty() ? "" : scenario.getRecentExecutions().get(0).getCurrentVersion().getName());
+                        renderVersionHeader(scenario.getHistoryExecutions().isEmpty() ? "" : scenario.getHistoryExecutions().get(0).getBaseVersion().getName());
+                        renderVersionHeader(scenario.getHistoryExecutions().isEmpty() ? "" : scenario.getHistoryExecutions().get(0).getCurrentVersion().getName());
                         th().text("Difference").end();
                         th().text("Confidence").end();
                     end();
-                    scenario.getRecentExecutions().forEach(execution -> {
+                    scenario.getHistoryExecutions().forEach(execution -> {
```

Comment:
How is it possible that the recent executions are empty?

Reply:
Because it's on a branch channel, there's only one "current execution" there.

Reply:
Yeah, and that one current execution is the one which should show up in the page, right? Or did I misunderstand something there?